spec/lib/tasks/eu_exit_business_finder_spec.rb

```diff
@@ -0,0 +1,179 @@
+      let(:facet_values) { ["000-remove-me"] }
+      let(:mock_behaviour!) { mock_facet_values_to_remove(["000-remove-me"]) }
+
+      it { is_expected.to eql [] }
+    end
+
+    context "given a mixture of values to keep and remove" do
+      let(:facet_values) { ["123-keep-me", "000-remove-me"] }
+      let(:mock_behaviour!) { mock_facet_values_to_remove(["000-remove-me"]) }
+
+      it { is_expected.to eql ["123-keep-me"] }
+    end
+
+    context "given values to replace" do
+      let(:facet_values) { ["000-old"] }
+      let(:mock_behaviour!) {
+        mock_facet_values_to_replace("000-old" => ["111-new"])
+      }
+
+      it { is_expected.to eql ["111-new"] }
+    end
+
+    context "given a mixture of values to replace, remove and keep" do
+      let(:facet_values) { ["000-old", "000-remove-me", "123-keep-me"] }
+      let(:mock_behaviour!) {
+        mock_facet_values_to_remove(["000-remove-me", "000-old"])
+        mock_facet_values_to_replace("000-old" => ["111-new"])
+      }
+
+      it { is_expected.to eql ["111-new", "123-keep-me"] }
+    end
+
+    context "given facet replacements which could lead to duplicates" do
+      let(:facet_values) { ["000-old", "001-also-old"] }
+      let(:mock_behaviour!) {
+        mock_facet_values_to_replace(
+          "000-old" => "111-new",
+          "001-also-old" => "111-new",
+        )
+      }
+
+      it { is_expected.to eql ["111-new"] } # `111-new` should only appear once
+    end
+
+    context "given a facet value which has been split into multiple new facets" do
+      let(:facet_values) { ["000-old"] }
+      let(:mock_behaviour!) {
+        mock_facet_values_to_replace(
+          "000-old" => ["111-new", "222-new"]
+        )
+      }
+
+      it { is_expected.to eql ["111-new", "222-new"] }
+    end
+  end
+
+  describe "updating the title" do
+    subject {
+      stub_publishing_api_has_item(
+        "content_id": "000-content-id-of-food-drink-tobacco-facet",
+        "title": "Food, drink and tobacco (retail and wholesale)",
+      )
+      stub_publishing_api_has_item(
+        "content_id": "111-content-id-of-ports-airports",
+        "title": "Ports and airports",
+      )
+      stub_publishing_api_has_item(
+        "content_id": "777-content-id-of-employing-eu-citizens",
+        "title": "EU citizens",
+      )
+      stub_publishing_api_has_item(
+        "content_id": "444-content-id-of-eu-funding",
+        "title": "EU funding",
+      )
+      mock_behaviour!
+      create_list_and_execute_rake!
+      SubscriberList.last.title
+    }
+
+    context "referencing a facet value in the finder data" do
+      let(:facet_values) { ["000-content-id-of-food-drink-tobacco-facet"] }
+
+      it {
+        is_expected.to eql "EU Exit guidance for your business in the following category: 'Food, drink and tobacco (retail and wholesale)'"
+      }
+    end
+
+    context "referencing multiple facet values in the finder data" do
```

Comment:
The same point about the test name applies here too.

How about: "build title from multiple facet values" ?